src/Admin/BaseFieldDescription.php

```diff
@@ -413,4 +433,54 @@ public function isVirtual()
     {
         return false !== $this->getOption('virtual_field', false);
     }
+
+    private function getFieldGetterKey($object, $fieldName)
+    {
+        return is_string($fieldName)
+            ? get_class($object).'-'.$fieldName
+                .(is_string($this->getOption('code'))
+                && '' !== $this->getOption('code')
```

Comment:
Is there no other way to do this, maybe separate into multiple private methods, @sonata-project/contributors wdyt? This looks very dirty to me.

Reply:
Might be clearer with an early return: 
```php
if (!is_string($fieldName)) {
    return null;
}
$keyComponents = [get_class($object), $fieldName];
$code = $this->getOption('code');
if (is_string($code) && $code !== '') {
    $keyComponents[] = $code;
}
return implode('-', $keyComponents);
```

Reply:
Thanks, looks better indeed